### DISCUSSION TITLE
usePrompt

### DISCUSSION THREAD

**Discussion by akraines** [TanStack/router#214] [Q&A] [answered]
Does react-location support the equivalent of use prompt that was removed form RR6?
https://github.com/remix-run/react-router/issues/8139#issue-1027969054

**Comment by tannerlinsley** [comment, marked as answer]
Yep

Tanner Linsley

**Reply by Zloka** [reply]
Cool! Does it allow to differentiate between actions like popping the navigation stack, and the user closing the tab? As I tried to outline in [this comment](https://github.com/remix-run/react-router/issues/7782#issuecomment-826583578), my use-case is such that I would like to prevent all "in-app navigation", but I don't have any reason to prevent the user from closing the tab.

Currently, my `usePrompt` usage asks the user if they are sure they wish to close the tab, which might spook them about unsaved changes unnecessarily.
